### DISCUSSION TITLE
lualine color and bubble theme

### DISCUSSION THREAD

**Discussion by xiehuc** [nanozuki/tabby.nvim#51] [Show and tell]
<img width="1322" alt="截屏2022-03-12 下午11 47 10" src="https://user-images.githubusercontent.com/571878/158024863-77f69f6f-7001-4cff-867e-09afec241c72.png">

```
--- https://github.com/nanozuki/tabby.nvim/blob/main/lua/tabby/presets.lua

local util = require('tabby.util')

local hl_tabline_fill = util.extract_nvim_hl('lualine_c_normal') -- 背景
local hl_tabline = util.extract_nvim_hl('lualine_b_normal')
local hl_tabline_sel = util.extract_nvim_hl('lualine_a_normal') -- 高亮

local function tab_label(tabid, active)
  local icon = active and '' or ''
  local number = vim.api.nvim_tabpage_get_number(tabid)
  local name = util.get_tab_name(tabid)
  return string.format(' %s %d: %s ', icon, number, name)
end

local presets = {
  hl = 'lualine_c_normal',
  layout = 'tab_only',
  head = {
    { '  ', hl = { fg = hl_tabline.fg, bg = hl_tabline.bg } },
    { '', hl = { fg = hl_tabline.bg, bg = hl_tabline_fill.bg } },
  },
  active_tab = {
    label = function(tabid)
      return {
        tab_label(tabid, true),
        hl = { fg = hl_tabline_sel.fg, bg = hl_tabline_sel.bg, style = 'bold' },
      }
    end,
    left_sep = { '', hl = { fg = hl_tabline_sel.bg, bg = hl_tabline_fill.bg } },
    right_sep = { '', hl = { fg = hl_tabline_sel.bg, bg = hl_tabline_fill.bg } },
  },
  inactive_tab = {
    label = function(tabid)
      return {
        tab_label(tabid, false),
        hl = { fg = hl_tabline.fg, bg = hl_tabline_fill.bg },
      }
    end,
    left_sep = { ' ', hl = { fg = hl_tabline.bg, bg = hl_tabline_fill.bg } },
    right_sep = { ' ', hl = { fg = hl_tabline.bg, bg = hl_tabline_fill.bg } },
  },
}

return presets

```

```
-- https://github.com/nvim-lualine/lualine.nvim/blob/master/examples/bubbles.lua
-- Bubbles config for lualine
-- Author: lokesh-krishna
-- MIT license, see LICENSE for more details.

-- stylua: ignore
local colors = {
  blue   = '#5E81AC',
  cyan   = '#88C0D0',
  black  = '#2E3440',
  white  = '#ECEFF4',
  red    = '#BF616A',
  violet = '#B48EAD',
  grey   = '#4C566A',
  green  = '#A3BE8C',
  orange = '#D08770',
}

local bubbles_theme = {
  normal = {
    a = { fg = colors.black, bg = colors.blue },
    b = { fg = colors.white, bg = colors.grey },
    c = { fg = colors.gery, bg = colors.black },
  },

  insert = { a = { fg = colors.black, bg = colors.green } },
  visual = { a = { fg = colors.black, bg = colors.violet } },
  replace = { a = { fg = colors.black, bg = colors.red } },

  inactive = {
    a = { fg = colors.white, bg = colors.black },
    b = { fg = colors.white, bg = colors.black },
    c = { fg = colors.black, bg = colors.black },
  },
}

require('lualine').setup({
  options = {
    theme = bubbles_theme,
    component_separators = '|',
    section_separators = { left = '', right = '' },
  },
  sections = {
    lualine_a = {
      { 'mode', separator = { left = '' }, right_padding = 2 },
    },
    lualine_b = { 'filename', 'branch' },
    lualine_c = { 'fileformat' },
    lualine_x = {},
    lualine_y = { 'filetype', 'progress' },
    lualine_z = {
      { 'location', separator = { right = '' }, left_padding = 2 },
    },
  },
  inactive_sections = {
    lualine_a = { 'filename' },
    lualine_b = {},
    lualine_c = {},
    lualine_x = {},
    lualine_y = {},
    lualine_z = { 'location' },
  },
  extensions = {},
})

```

**Comment by ciscohack** [comment]
I tried to use bubble_theme style for my lualine in init.lua and sadly it's giving me some error and I am not able to fix it .. I just copied the code and pasted in under below section I hope doing correct.

 'nvim-lualine/lualine.nvim',
    'nvim-tree/nvim-web-devicons',
    -- See `:help lualine.txt`
    opts = {
    --   options = {
        icons_enabled = true,
      },
    config = function()

**Reply by xiehuc** [reply]
what error?

**Reply by ciscohack** [reply]
some syntax and function errors were showing ..Since I was not able to fix so removed it

**Comment by xiehuc** [comment]
update demo

lazy.nvim

```
  {
    'nanozuki/tabby.nvim',
    dependencies = { 'nvim-lualine/lualine.nvim' },
    config = function()
      require('tabby').setup({
        tabline = require('tabby-cfg'),
      })
    end
  },
```

tabby-cfg.lua
```
--- https://github.com/nanozuki/tabby.nvim/blob/main/lua/tabby/presets.lua

local util = require('tabby.util')

local hl_tabline_fill = util.extract_nvim_hl('lualine_c_normal') -- 背景
local hl_tabline = util.extract_nvim_hl('lualine_b_normal')
local hl_tabline_sel = util.extract_nvim_hl('lualine_a_normal')  -- 高亮

local function tab_label(tabid, active)
  local icon = active and ' ' or ' '
  local number = vim.api.nvim_tabpage_get_number(tabid)
  local name = util.get_tab_name(tabid)
  return string.format(' %s %d: %s ', icon, number, name)
end

local presets = {
  hl = 'lualine_c_normal',
  layout = 'tab_only',
  head = {
    { '  ', hl = { fg = hl_tabline.fg, bg = hl_tabline.bg } },
    { '', hl = { fg = hl_tabline.bg, bg = hl_tabline_fill.bg } },
  },
  active_tab = {
    label = function(tabid)
      return {
        tab_label(tabid, true),
        hl = { fg = hl_tabline.fg, bg = hl_tabline_sel.bg },
      }
    end,
    left_sep = { '', hl = { fg = hl_tabline_sel.bg, bg = hl_tabline_fill.bg } },
    right_sep = { '', hl = { fg = hl_tabline_sel.bg, bg = hl_tabline_fill.bg } },
  },
  inactive_tab = {
    label = function(tabid)
      return {
        tab_label(tabid, false),
        hl = { fg = hl_tabline.fg, bg = hl_tabline_fill.bg },
      }
    end,
    left_sep = { ' ', hl = { fg = hl_tabline.bg, bg = hl_tabline_fill.bg } },
    right_sep = { ' ', hl = { fg = hl_tabline.bg, bg = hl_tabline_fill.bg } },
  },
}

return presets
```

**Reply by xiehuc** [reply]
need nerdfont

**Comment by ciscohack** [comment]
> update demo
> 
> lazy.nvim
> 
> ```
>   {
>     'nanozuki/tabby.nvim',
>     dependencies = { 'nvim-lualine/lualine.nvim' },
>     config = function()
>       require('tabby').setup({
>         tabline = require('tabby-cfg'),
>       })
>     end
>   },
> ```
> 
> tabby-cfg.lua
> 
> ```
> --- https://github.com/nanozuki/tabby.nvim/blob/main/lua/tabby/presets.lua
> 
> local util = require('tabby.util')
> 
> local hl_tabline_fill = util.extract_nvim_hl('lualine_c_normal') -- 背景
> local hl_tabline = util.extract_nvim_hl('lualine_b_normal')
> local hl_tabline_sel = util.extract_nvim_hl('lualine_a_normal')  -- 高亮
> 
> local function tab_label(tabid, active)
>   local icon = active and ' ' or ' '
>   local number = vim.api.nvim_tabpage_get_number(tabid)
>   local name = util.get_tab_name(tabid)
>   return string.format(' %s %d: %s ', icon, number, name)
> end
> 
> local presets = {
>   hl = 'lualine_c_normal',
>   layout = 'tab_only',
>   head = {
>     { '  ', hl = { fg = hl_tabline.fg, bg = hl_tabline.bg } },
>     { '', hl = { fg = hl_tabline.bg, bg = hl_tabline_fill.bg } },
>   },
>   active_tab = {
>     label = function(tabid)
>       return {
>         tab_label(tabid, true),
>         hl = { fg = hl_tabline.fg, bg = hl_tabline_sel.bg },
>       }
>     end,
>     left_sep = { '', hl = { fg = hl_tabline_sel.bg, bg = hl_tabline_fill.bg } },
>     right_sep = { '', hl = { fg = hl_tabline_sel.bg, bg = hl_tabline_fill.bg } },
>   },
>   inactive_tab = {
>     label = function(tabid)
>       return {
>         tab_label(tabid, false),
>         hl = { fg = hl_tabline.fg, bg = hl_tabline_fill.bg },
>       }
>     end,
>     left_sep = { ' ', hl = { fg = hl_tabline.bg, bg = hl_tabline_fill.bg } },
>     right_sep = { ' ', hl = { fg = hl_tabline.bg, bg = hl_tabline_fill.bg } },
>   },
> }
> 
> return presets
> ```

Thanks for giving me the snippet of config let me try

**Comment by ciscohack** [comment]
tried new config getting module not found though tabby.vim downloaded


<img width="731" alt="image" src="https://github.com/nanozuki/tabby.nvim/assets/11479557/bdb8e15e-8a62-4bee-bf1c-62879f46b271">


<img width="1060" alt="image" src="https://github.com/nanozuki/tabby.nvim/assets/11479557/2114c586-a14a-4754-a4c1-722df3e42594">
